rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -264,6 +264,19 @@ public RMCore(String id, Node nodeRM) throws ActiveObjectCreationException, Node
         this.jmxHelper = new RMJMXHelper(this.accountsManager);
     }
 
+    public RMCore(HashMap<String, NodeSource> nodeSources, ArrayList<String> brokenNodeSources,
+                  HashMap<String, RMNode> allNodes, Client caller, RMMonitoringImpl monitoring,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The type of the "allNodes" object should be an interface such as "Map" rather than the implementation "HashMap". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1319)